src/main/java/org/motechproject/mots/web/UserController.java

```diff
@@ -86,6 +88,25 @@ public UserDto saveUser(@PathVariable("id") UUID id, @RequestBody UserDto userDt
     return userMapper.toDto(userService.saveUser(existingUser));
   }
 
+  /**
+   * Update User password.
+   * @param oldPassword user previous password, should be the same as current password
+   * @param newPassword user's new password
+   */
+  @RequestMapping(value = "/user/password/change",
+      method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public String changeCurrentUserPassword(@RequestParam("oldPassword") String oldPassword,
+      @RequestParam("newPassword") String newPassword) {
```

Comment:
Did you try to add Principal as a parameter here?
public String changeCurrentUserPassword(@RequestParam("oldPassword") String oldPassword, @RequestParam("newPassword") String newPassword, Principal principal)